src/Microsoft.ML.Core/Utilities/NormStr.cs

```diff
@@ -116,7 +116,7 @@ public NormStr Get(string str, bool add = false)
                 return add ? AddCore(str.AsMemory(), hash) : null;
             }
 
-            public NormStr Get(ReadOnlyMemory<char> str, bool add = false)
+            public NormStr Get(ReadOnlyMemory<char> str, bool add = false, bool createNewStr = true)
```

Comment:
You'll want to expose the same flag in the Add() methods, and any other callers of this function within the NormStr class.

Speaking of, you may want to investigate which outside components call this interface to see if their memory usage / runtime will increase when defaulting to `true`.

Reply:
The only method inside the `NormStr` class that calls `Get()` with `add = true` is the Add() method you mention. So that's going to be the only other method where I will add the flag.

About your other suggestion, I will discuss with @harishsk about where to investigate. But it seems to me that the Add() method is _only_ used in 2 other classes:
1. The `ValueToKeyMapperTransformer.Builder.TextImpl` , which is the class that holds the _pool that I've been working with in this issue. It might be worth exploring cases where the VTKMT would have a worse memory usage with the changes introduced in this PR. I've tried with a couple of different datasets, but I haven't seen this being an issue, and I don't know which kind of cases to explore.
2. The `ModelSaveContext`, where strings are serialized to disk as indices, and they're added to a NormStr.Pool that is later serialized. In this case I don't see why the changes in this PR could cause a problem in its memory usage/runtime (because the changes of this PR would only execute once, when the `str` is not already in the pool).

Reply:
Thanks for checking for other uses which would be affected.